dataprofiler/profilers/profile_builder.py

```diff
@@ -683,3 +687,127 @@ def tqdm(l):
+        if data_labeler_options.is_enabled \
+                and data_labeler_options.data_labeler_object is None:
+            try:
+                data_labeler = DataLabeler(
+                    labeler_type='structured',
+                    dirpath=data_labeler_options.data_labeler_dirpath,
+                    load_options=None)
+                self.options.set({'data_labeler.data_labeler_object': data_labeler})
+                
+            except Exception as e:
+                utils.warn_on_profile('data_labeler', e)
+                self.options.set({'data_labeler.is_enabled': False})
+                
+        # Restore data labelers for all columns
+        for key in self._profile:
+
+            val = self._profile[key]
+            use_data_labeler = True
+            if val.options and isinstance(val.options, StructuredOptions):
+                use_data_labeler = val.options.data_labeler.is_enabled
+
+            if use_data_labeler:
+                data_labeler_profile = val.profiles['data_label_profile']._profiles['data_labeler']
+                data_labeler_profile.data_labeler = None
+                
+                if val.options and val.options.data_labeler.data_labeler_object:
+                    data_labeler_profile.data_labeler = val.options.data_labeler.data_labeler_object
+                if data_labeler_profile.data_labeler is None:
+                    data_labeler_dirpath = None
+                    if val.options:
+                        data_labeler_dirpath = val.options.data_labeler.data_labeler_dirpath
+                
+                    data_labeler_profile.data_labeler = DataLabeler(
+                        labeler_type='structured',
+                        dirpath=data_labeler_dirpath,
+                        load_options=None)
+
+
+    def save(self, filepath="profile.pkl"):
+        """
+        Save profiler to disk
+        
+        :param filepath: Path of file to save to
+        :type filepath: String
+        :return: None
+        """
+        # Delete data labelers as they can't be pickled
+        self._delete_data_labelers()
+
+        # Create dictionary for all metadata, options, and profile 
+        data = { 
+                "total_samples": self.total_samples,
+                "encoding": self.encoding,
+                "file_type": self.file_type,
+                "row_has_null_count": self.row_has_null_count,
+                "row_is_null_count": self.row_is_null_count,
+                "hashed_row_dict": self.hashed_row_dict,
+                "_samples_per_update": self._samples_per_update,
+                "_min_true_samples": self._min_true_samples,
+                "options": self.options,
+                "_profile": self.profile
+               } 
+
+        # Pickle and save profile to disk
+        with open(filepath, "wb") as outfile:
+            pickle.dump(data, outfile)
+
+        # Restore all data labelers
+        self._restore_data_labelers()
```

Comment:
maybe I misunderstand your code, but why we need to delete labeler first, and then restore it at the end when we save the profile.

Reply:
The Data Labeler objects were deleted because they can't be pickled. That is why I added both a delete and restore function. The restore function is called after saving as well so the data_labeler_object can still be used after saving the profile.

Reply:
interesting. Then, we should only save other components except data labeler (the data labeler by default is just to predict, not supposed to be updated, so the resource folder should be good to load it when needed). We should somehow bundle with the resource containing data labeler to a zip file or something.

Then, when we load the profiler, we load all other components, along with the data labeler loaded from the resource in that zip file.
Any thought?

Reply:
@lettergram @JGSweets

Reply:
I don't think a labeler should be bundled with a profile as that would be a lot of memory. We should just save references to where that labeler exists.

Reply:
That's kind of my thought process. Saving the reference to data labeler is sufficient if we work on the same machine. If we copy the saved profiler to another machine, I think we need to bundle all components, but I'm not sure we should tackle that usage for now.

Reply:
you only need bundled components if it isn't part of the library itself though

Reply:
Also, we might want to save profiler versions with the save

Reply:
@AnhTruong As far as saving the references, the way I am "deleting" and "restoring" is just to set the references to None and then use the file paths and reload them based on whether those options were set or not. Is the current solution sufficient?